### PR TITLE
cquery v20180302 (new formula)

### PR DESCRIPTION
**Pull request by xu-cheng**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by xu-cheng** [comment]
Any comments?

File: Formula/cquery.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Cquery < Formula
+  desc "C/C++ language server"
+  homepage "https://github.com/cquery-project/cquery"
+  # pull from git tag to get submodules
+  url "https://github.com/cquery-project/cquery.git", :tag => "v20180302",
+                                                      :revision => "f3e9e756e182b122bef8826a77047f6ccf5529b6"
+  head "https://github.com/cquery-project/cquery.git"
```
Comment (fxcoudert) [inline Formula/cquery.rb]: No `head` for new formulas

File: Formula/cquery.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  def install
+    system "./waf", "configure", "--prefix=#{prefix}"
+    system "./waf", "build"
+    system "./waf", "install"
+  end
+
+  test do
+    system bin/"cquery", "--test-unit"
```
Comment (fxcoudert) [inline Formula/cquery.rb]: Can you tell us what that test does?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/cquery.rb (hunk 2 above)
Comment (tycho) [reply Formula/cquery.rb]:
```
$ cquery --test-unit
[doctest] doctest version is "1.2.7"
[doctest] run with "--help" for options
===============================================================================
[doctest] test cases:     53 |     53 passed |      0 failed |      0 skipped
[doctest] assertions:    326 |    326 passed |      0 failed |
[doctest] Status: SUCCESS!
```
So, numerous unit tests.

File: Formula/cquery.rb (hunk 1 above)
Comment (xu-cheng) [reply Formula/cquery.rb]: Could we keep it? `cquery` is a relatively new project and is seeing rapidly development. Therefore, there is a need for certain users to follow the HEAD.

Comment (tycho) [reply Formula/cquery.rb]: FWIW, I agree wiht @xu-cheng. Having HEAD woul dbe a good idea. Why is there a ban on head for new formulae?

Comment (fxcoudert) [reply Formula/cquery.rb]: Because it tends to break often, and create more work for us.

Comment (tycho) [reply Formula/cquery.rb]: Add a big disclaimer, like: "YOU'RE BUILDING FROM HEAD. IF IT BREAKS, DEAL WITH IT. WE WON'T HELP YOU. (Patches welcome.)"

Comment (fxcoudert) [reply Formula/cquery.rb]: But then patches have to be reviewed, and merged, by someone 😈 
I'd tend to say: if you need to build from head, convince upstream that they should release more often!

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by tycho** [comment]
Thanks to each of you for getting this one merged -- been looking forward to this one.

**Pull request closed without merging**